contracts/src/protocol/IPTokenStaking.sol

```diff
@@ -455,13 +426,19 @@ contract IPTokenStaking is IIPTokenStaking, Ownable2StepUpgradeable, ReentrancyG
         if (stakeAmount < minStakeAmount) {
             revert Errors.IPTokenStaking__StakeAmountUnderMin();
         }
-        uint32 duration = 0;
         uint256 delegationId = 0;
         if (stakingPeriod != IIPTokenStaking.StakingPeriod.FLEXIBLE) {
```

Comment:
Can we verify that stakingPeriod has values that are accepted by the `enum StakingPeriod` ?

Reply:
Good point

Reply:
Fixed 71d5c45